LogMonitor/src/LogMonitor/EtwMonitor.h

```diff
@@ -10,6 +10,23 @@ typedef LPTSTR(NTAPI* PIPV6ADDRTOSTRING)(
     LPTSTR S
     );
 
+//
+// struct to hold the ETW logEntry data
+//
+struct EtwLogEntry {
+    std::wstring Time;
+    std::wstring ProviderId;
+    std::wstring ProviderName;
+    std::wstring DecodingSource;
+    int ExecProcessId;
+    int ExecThreadId;
+    std::wstring Level;
+    std::wstring Keyword;
+    std::wstring EventId;
+    std::vector<std::pair<std::wstring, std::wstring>> EventData{};
+};
+
+

```

Comment:
ETW can be instructed by many sources i.e XML files (manifest instrumentation), WPP, MOF classes etc. Will this capture the fields generated by all those versions?

Reply:
Yes, this is just a transformation of what was already there. I think we're mostly covering for XML and MOF classes from what the current code.